core/controllers/editor.py

```diff
@@ -706,6 +706,30 @@ def get(self, exploration_id, exploration_version):
             exploration_id, exploration_version))
 
 
+class HintStatisticsHandler(EditorHandler):
+    """Returns statistics for hints in a state."""
```

Comment:
I think you should get the stats at the exploration level, and retrieve them all just once at the outset when the editor page is loaded, rather than making ping after ping to the backend as the creator browses through the exploration while editing it. The statistics should be stored in such a way that they can be retrieved by exp_id and exp_version; this may entail some changes to the storage model definition.